Rlog/View/Components/CustomTextField.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  CustomTextField.swift
+//  MacCPlayGround
+//
+//  Created by Noah's Ark on 2022/10/19.
+//
+
+import SwiftUI
+
+enum CustomTextFieldType: String {
+    case workplace = "근무지"
+    case wage = "시급"
+    case payday = "급여일"
+    case reason = "사유"
+    case none = ""
+    
+    var placeholder: String {
+        switch self {
+        case .workplace: return "예시) 편의점"
+        case .wage: return "최저시급 9,160원"
+        case .payday: return "10"
+        case .reason: return "사유를 입력해주세요."
+        case .none: return "내용을 입력해주세요."
+        }
+    }
+}
+
+struct CustomTextField: View {
+    let textFieldType: CustomTextFieldType
+    let keyboardType: UIKeyboardType
+    @State var isFocused: Bool = false
+    @Binding var text: String {
+        didSet {
+            if textFieldType == .workplace {
+                if text.count > 20 && oldValue.count <= 20 {
+                    text = oldValue
+                }
+            } else if textFieldType == .wage {
+                if text.hasPrefix("0") { text = "" }
+            } else if textFieldType == .payday {
+                if text.hasPrefix("0") { text = "" }
+                if Int(text) ?? 10 > 31 || Int(text) ?? 10 < 1 { text = "" }
+            }
```

Comment:
```suggestion
            switch textFieldType {
            case .workplace:
                if text.count > 20 && oldValue.count <= 20 {
                    text = oldValue
                }
            case .wage:
                if text.hasPrefix("0") { text = "" }
            case .payday:
                if text.hasPrefix("0") { text = "" }
                if text.count > 31 || text.count < 1 { text = "" }
            default:
                return
            }
```

enum의 분기에서는 if문 대신 switch문을 사용하면 가독성이 좋아져서 추천드려요 ~ !!